lib/IRGen/IRGenMangler.h

```diff
@@ -16,6 +16,7 @@
 #include "IRGenModule.h"
 #include "swift/AST/ASTMangler.h"
 #include "swift/IRGen/ValueWitness.h"
+#include "llvm/Support/SaveAndRestore.h"
```

Comment:
Is this needed?

Reply:
Yes, it's used so we don't drop the protocol name in the mangling.

Reply:
Hmm. I could mangle this as <protocol> <assoc-type-name> and let it drop the associated type names from the rest of the mangling.